src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,102 @@
+
 public class WorkWithFile {
+    private static final int CONST_VALUE = 0;
+    private static final int STEP_SIZE = 2;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String data = readFile(fromFileName);
+        String report = createReport(data);
+        writeToFile(toFileName, report);
+    }
+
+    public String readFile(String fromFileName) {
+        StringBuilder stringBuilder = new StringBuilder();
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fromFileName))) {
+            String readLine = bufferedReader.readLine();
+            while (readLine != null) {
+                String[] matchLines = readLine.split("\\W+");
+                for (String word : matchLines) {
+                    stringBuilder.append(word).append(" ");
+                }
+                readLine = bufferedReader.readLine();
+            }
+        } catch (FileNotFoundException e) {
+            throw new RuntimeException("file" + fromFileName + "doesn't exist" + e);
+        } catch (IOException e) {
+            throw new RuntimeException("Can`t read file " + fromFileName + " " + e);
+        }
+        return stringBuilder.toString();
+    }
+
+    public String createReport(String report) {
+        StringBuilder stringBuilder = new StringBuilder();
+
+        String[] parts = report.split("\\W+");
+        String[] names = new String[parts.length / STEP_SIZE];
+        int[] sums = new int[parts.length / STEP_SIZE];
+
+        int size = CONST_VALUE;
+
+        for (int i = 0; i < parts.length; i += STEP_SIZE) {
+            String name = parts[i];
+            int value = Integer.parseInt(parts[i + 1]);
+
+            boolean nameFound = false;
+            for (int j = CONST_VALUE; j < size; j++) {
+                if (names[j] != null && names[j].equals(name)) {
+                    sums[j] += value;
+                    nameFound = true;
+                    break;
+                }
+            }
+            if (!nameFound) {
+                names[size] = name;
+                sums[size] = value;
+                size++;
+            }
+        }
+        for (int i = CONST_VALUE; i < sums.length; i++) {
+            String contName = names[i];
+            int contSum = sums[i];
+            for (int j = i + 1; j < sums.length; j++) {
+                if (sums[i] < sums[j]) {
+                    names[i] = names[j];
+                    names[j] = contName;
+
+                    sums[i] = sums[j];
+                    sums[j] = contSum;
+                }
+            }
+        }
+
+        for (int i = CONST_VALUE; i < size; i++) {
+            stringBuilder.append(names[i]).append(",").append(sums[i])
+                    .append(System.lineSeparator());
+        }
+
+        int sumResult = sums[CONST_VALUE];
+
+        for (int i = 1; i < sums.length; i++) {
+            sumResult -= sums[i];
+        }
+        stringBuilder.append("result,").append(sumResult);
```

Comment:
I really cannot get idea why we need 4 loops to solve the task
we can introduce 2 int variables for 2 possible operations (supply and buy)
then we go within 1 loop through rows of our data
if operation is buy - increasing one variable, if supply - other one.
After that we have all we need (2 values) and can form output with single StringBuilder
Something like
`return new StringBuilder()
                .append(SUPPLY).append(",").append(supplyCounter).append(System.lineSeparator())
                .append(BUY).append(",").append(buyCounter).append(System.lineSeparator())
                .append(RESULT).append(",").append(supplyCounter - buyCounter).toString();`